src/components/lobby/MatchItem.js

```diff
@@ -0,0 +1,150 @@
+import { leaveGame } from "utils/game/leave";
+import LeaveButton from "components/game/LeaveButton";
+
+import styles from "./MatchItem.module.css";
+
+const PlayersCounter = ({ count, total }) => (
+  <span className={styles.players}>
+    {count}/{total}
+    <Icon name="male" className={styles.playersIcon} />
+  </span>
+);
+
+const JoinGameButton = ({ current, isFull, canJoin, ...props }) => {
+  const { t } = useTranslation("lobby", "translation");
+
+  const buttonText = () => {
+    if (current) return t("list.game.play");
+    if (isFull) return t("list.game.full");
+    return t("list.game.join");
+  };
+
+  const buttonColor = () => {
+    if (current) return "green";
+    if (isFull) return "grey";
+    return "yellow";
+  };
+
+  return (
+    <Button {...props} className={styles.button} disabled={!canJoin} color={buttonColor()}>
+      {buttonText()}
+    </Button>
+  );
+};
+
+const matchItemPropTypes = {
+  match: MatchType.isRequired,
+  game: GameType,
+  onJoin: PropTypes.func.isRequired,
+  current: PropTypes.bool,
+  disabled: PropTypes.bool,
+};
+
+const MatchesItem = React.memo(
+  ({
+    match,
+    match: { matchID, gameName, players, setupData },
+    game,
+    onJoin,
+    current,
+    disabled,
+  }) => {
+    const { t } = useTranslation(["translation", "info"]);
+    const history = useHistory();
+    const user = useUser();
+    const profiles = useProfiles();
+    const credentials = useCredentials();
+    const [open, setOpen] = useState(false);
+
+    if (!game) return null;
+
+    const gameCredentials = credentials && credentials[matchID];
+    const currentPlayers = players.filter((p) => p.name);
+    const maxPlayers = players.length;
+    const isFull = currentPlayers.length === maxPlayers;
+    const isPrivate = setupData && setupData.private;
+    const canJoin = !disabled && (!isFull || current);
+
+    const handleJoin = () => {
+      if (canJoin) onJoin(match);
+    };
+
+    const handleLeave = () => {
+      leaveGame(
+        gameName,
+        matchID,
+        currentPlayers.find(({ name }) => name === user.uid).id.toString(),
+        user.uid,
+        gameCredentials,
+        history
+      );
+    };
+
+    const playerNames = currentPlayers.map(({ name: uid }) => {
+      const profile = profiles.get(uid);
+      return profile ? profile.displayName : uid;
+    });
+
+    return (
+      <div className={classNames(styles.matchItem, { [styles.detailsOpen]: open })}>
+        <div className={styles.row}>
+          <span className={styles.showDetails} onClick={() => setOpen((open) => !open)}>
+            {open
+              ? `${t("lobby:list.game.less_details")} ▲`
+              : `${t("lobby:list.game.more_details")} ▼`}
+          </span>
+
+          <img className={styles.image} src={game.image} alt="game icon" />
+          <h3 className={styles.gameName}>{game.displayName || game.name}</h3>
+          <h4 className={styles.matchName}>Super game room</h4>
```

Comment:
Can we actually allow setting custom match names?

Reply:
Yes, but I'm adding this as a separate PR

Reply:
I forgot to push a commit that displays `#{matchID}` instead, until the custom match name feature is implemented